src/cmd/launch.rs

```diff
@@ -22,3 +123,171 @@ impl LaunchStruct {
+                None => continue,
+            },
+            None => continue,
+        };
+
+        fs::copy(
+            &path,
+            format!(
+                "{}/{}/upper/{}",
+                launch.user().containers(),
+                launch.name(),
+                file_name
+            ),
+        )?;
+    }
+
+    Ok(())
+}
+
+fn remount<DO: Downloader>(launch: &command::Launch<DO>) -> Result<(), Box<dyn std::error::Error>> {
+    let rootfs_path = match launch.rootfs_option() {
+        RootFSOption::RootfsImage(image) => {
+            match image.check_rootfs_newest() {
+                Ok(flg) => match flg {
+                    true => { /* do nothing */ }
+                    false => {
+                        image.download_image()?;
+                    }
+                },
+                Err(e) => {
+                    return Err(e)?;
+                }
+            }
+            image.rootfs_path()
+        }
+        _ => return Err(Error::NotImplemented)?,
+    };
+    let lower_dir = launch
+        .target_container()
+        .lowerdir()
+        .to_str()
+        .ok_or(Error::LowerDirNotFound)?;
+    let upper_dir = launch
+        .target_container()
+        .upperdir()
+        .to_str()
+        .ok_or(Error::UpperDirNotFound)?;
+    let work_dir = launch
+        .target_container()
+        .workdir()
+        .to_str()
+        .ok_or(Error::WorkDirNotFound)?;
+
+    // デバック対象コンテナのlowerdirに対してrootfsを追加した後reloadする
+    umount2(launch.target_container().mergeddir(), MntFlags::empty())
+        .map_err(|why| Error::UnmountFailed(why))?;
+
+    mount(
+        Some("overlay"),
+        // launch.target_container().mergeddir(),
+        launch.target_container().mergeddir(),
+        Some("overlay"),
+        MsFlags::empty(),
+        Some(
+            format!(
+                "lowerdir={}:{},upperdir={},workdir={}",
+                rootfs_path.to_str().ok_or(Error::InvalidRootFSPath)?,
+                lower_dir,
+                upper_dir,
+                work_dir,
+            )
+            .as_str(),
+        ),
+    )
+    .map_err(|why| Error::MountFailed(why))?;
+
+    Ok(())
+}
+
+/// プロセスのnamespace不ァイルディスクリプタを管理する構造体
+struct Ns {
+    net: File,
+    cgroup: File,
+    ipc: File,
+    pid: File,
+    user: File,
+    mnt: File,
+    uts: File,
+}
+
+impl Ns {
+    fn new(container_pid: u32) -> Result<Ns, Box<dyn std::error::Error>> {
+        Ok(Ns {
+            net: File::open(format!("/proc/{}/ns/net", container_pid))?,
+            cgroup: File::open(format!("/proc/{}/ns/cgroup", container_pid))?,
+            ipc: File::open(format!("/proc/{}/ns/ipc", container_pid))?,
+            pid: File::open(format!("/proc/{}/ns/pid", container_pid))?,
+            user: File::open(format!("/proc/{}/ns/user", container_pid))?,
+            mnt: File::open(format!("/proc/{}/ns/mnt", container_pid))?,
+            uts: File::open(format!("/proc/{}/ns/uts", container_pid))?,
+        })
```

Comment:
基となるpathを定義しておく
openは&strでいけるっぽいので、わざわざpathにしなくていいかな…？

```suggestion
        let ns_base_path = format!("/proc/{}/ns", container_pid);
        Ok(Ns {
            net: File::open(format!("{}/net", &ns_base_path))?,
            cgroup: File::open(format!("{}/cgroup", &ns_base_path))?,
            ipc: File::open(format!("{}/ipc", &ns_base_path))?,
            pid: File::open(format!("{}/pid", &ns_base_path))?,
            user: File::open(format!("{}/user", &ns_base_path))?,
            mnt: File::open(format!("{}/mnt", &ns_base_path))?,
            uts: File::open(format!("{}/uts", &ns_base_path))?,
        })
```